public/app/plugins/datasource/prometheus/datasource.tsx

```diff
@@ -237,6 +237,7 @@ export class PrometheusDatasource
       this._request<T>(`/api/datasources/${this.id}/resources${url}`, params, {
         method: 'GET',
         hideFromInspector: true,
+        showErrorAlert: false,
```

Comment:
This is set to showErrorAlert: false, matching the same call in the if clause above.

This suppresses the errors from the following paths:
`/api/v1/rules`
`/api/v1/metadata`

Reply:
So I am just unsure here because it means that if there is any issue with labels/values API calls this will also be suppressed and user will end up with queryEditor that is failing without any indication. If we need it only for /rules and /metadata can we lift it up? Or can we just handle the error manually and look if it is a problem with multiple tenants and only hide it then?

Reply:
Hmmm... true. And if we can't fetch the labels/values, it won't be immediately clear that there is a problem unless they manually write queries and execute them?

Checking for the multi-tenants case would require some backend insight. Also, is it possible to suppress the error when catching it, or is it too late?

This closed PR suggests adding a parameter (see: https://github.com/grafana/grafana/pull/52249) -- then we ensure that we only suppress on `/metadata` and `/rules`. Except, I think I'd prefer something similar to @leventebalogh's and add a parameter for `options?: Partial<BackendSrvRequest>`.

Reply:
@aocenas , what is your thought on the previous "IF" block, which sets `showErrorAlert` to false already:

(See line 223 above)
```typescript
  async metadataRequest<T = any>(url: string, params = {}) {
    // If URL includes endpoint that supports POST and GET method, try to use configured method. This might fail as POST is supported only in v2.10+.
    if (GET_AND_POST_METADATA_ENDPOINTS.some((endpoint) => url.includes(endpoint))) {
      try {
        return await lastValueFrom(
          this._request<T>(`/api/datasources/${this.id}/resources${url}`, params, {
            method: this.httpMethod,
            hideFromInspector: true,
            showErrorAlert: false, // <-- this is line 223
          })
        );
      } catch (err) {
        // If status code of error is Method Not Allowed (405) and HTTP method is POST, retry with GET
        if (this.httpMethod === 'POST' && isFetchError(err) && (err.status === 405 || err.status === 400)) {
          console.warn(`Couldn't use configured POST HTTP method for this request. Trying to use GET method instead.`);
        } else {
          throw err;
        }
      }
    }
    return await lastValueFrom(
      this._request<T>(`/api/datasources/${this.id}/resources${url}`, params, {
        method: 'GET',
        hideFromInspector: true,
        showErrorAlert: false, // <-- Current line we are discussing
...        
```

Reply:
> And if we can't fetch the labels/values, it won't be immediately clear that there is a problem unless they manually write queries and execute them?

Well, I think the user just won't get any autocomplete in the code editor or no options in any dropdowns in the editor. They will be just empty and if we suppress the error there is not much other indication. So it's probably rare situation but still not great not to have any message there.

So I sort of like
> This closed PR suggests adding a parameter (see: #52249) -- then we ensure that we only suppress on /metadata and /rules. Except, I think I'd prefer something similar to @leventebalogh's #51316 and add a parameter for options?: Partial<BackendSrvRequest> to metadataRequest.

for this reason, as it seems like a smaller change/less of a regression at this point.

> what is your thought on the previous "IF" block, which sets showErrorAlert to false already:

So this might need a better comment in code but the reason it is there is that at first, we do a POST request, which may fail on older Prom instances because POST is supported only from a specific version. So we try the POST and if it fails in a specific way we retry with GET. So that is why we want to handle any error from the first request manually because we can potentially recover from it, and if the second fails we just show the default message because we have nothing else to do.